polybuild/src/build.d

```diff
@@ -126,12 +126,46 @@ int buildApp(ref Buildfile buildfile, ref PolybuildOptions options)
         objFiles ~= "build/d_monolithic_obj_file.o";
     }
 
+    if (buildfile.allSources.languages.zig)
+    {
+        // We'll start by building a list of the files we need to copy.
+        string[] wrappersInBuildTree;
+        foreach (file; buildfile.generatedSources.zigSources)
+        {
+            wrappersInBuildTree ~= file;
+        }
+
+        // ..and compile them.
+        foreach (file; buildfile.allSources.zigSources)
+        {
+            string objFile = "build/" ~ file ~ ".o";
+            // fcompiler-rt - zig have own compiler-rt (LLVM-compiler-rt rewritten on Zig [stage2])
+            // replacing stack-protector to zig-stack-protector
+            auto command = ["zig", "build-obj", file, "-fcompiler-rt", "-lc++", "-femit-bin=" ~ objFile];
+            if (options.verbose)
+                writeln("Executing " ~ command.join(' '));
+            retval = spawnProcess(command).wait();
+            if (retval != 0)
+                return retval;
+
+            objFiles ~= objFile;
+        }
+    }
+
     if (options.verbose)
         writeln("Linking " ~ objFiles.sort.uniq.array.to!string);
-    retval = spawnProcess(["clang++"] ~ objFiles.sort.uniq.array ~ [
-        "-lphobos2-ldc-shared", "-ldruntime-ldc-shared",
-        getRustStandardLibraryPath(), "-o", "build/" ~ buildfile.projectName
-    ]).wait();
-
+    if (buildfile.allSources.languages.d)
```

Comment:
I'd prefer if you would create an array of linker flags here and then call clang only once.

Reply:
commit: https://github.com/LorenDB/polyglot/pull/1/commits/42572a38f46dd0d4d042306f81c2ee91790d6933